src/Nethermind/Nethermind.Optimism/CL/P2P/P2PBlockValidator.cs

```diff
@@ -86,7 +86,7 @@ private bool IsTimestampValid(ExecutionPayloadV3 payload)
     private bool IsBlockHashValid(ExecutionPayloadV3 payload)
     {
         // [REJECT] if the block_hash in the payload is not valid
-        payload.TryGetBlock(out Block? block);
+        Block? block = payload.TryGetBlock().Block;
```

Comment:
I see we are basically dropping the underlying error here and in some other places, and it doesn't look like it's guaranteed  to be logged when `*Result` is created. Is this ok?